cmd/apiserver/server.go

```diff
@@ -60,12 +57,77 @@ func (s *server) createCalculationBulk(c *gin.Context) {
 	}
 
 	if err := s.client.Create(s.ctx, bulk); err != nil {
-		responseError(c, "couldn't create calculation", err)
+		responseError(c, "couldn't create calculation bulk", err)
 	} else {
 		c.JSON(http.StatusOK, gin.H{"data": bulk})
 	}
 }
 
+func (s *server) createWorkerPool(c *gin.Context) {
+	workerPoolName := c.Query("name")
+
+	workerPool := &workersv1.WorkerPool{
+		ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("workerpool-%s", workerPoolName), Namespace: s.namespace},
+	}
+
+	s.logger.Infof("Creating the workerpool %s...", workerPool.Name)
+
+	if err := s.client.Create(s.ctx, workerPool); err != nil {
+		responseError(c, "couldn't create workerpool", err)
+		return
+	}
+
+	newWorkerPool := &workersv1.WorkerPool{}
+	if err := s.client.Get(s.ctx, ctrlruntimeclient.ObjectKey{Namespace: workerPool.GetNamespace(), Name: workerPool.GetName()}, newWorkerPool); err != nil {
```

Comment:
we should retry on conflict here as we do in the rest of the times we get an object.